src/subcommands/util.rs

```diff
@@ -304,45 +317,21 @@ message = "0x"
                 });
                 Ok(resp.render(format, color))
             }
-            ("sign-data", Some(m)) => {
-                let binary: Vec<u8> = HexParser.from_matches(m, "binary-hex")?;
-                let recoverable = m.is_present("recoverable");
-                let from_privkey_opt: Option<PrivkeyWrapper> =
-                    PrivkeyPathParser.from_matches_opt(m, "privkey-path", false)?;
-                let from_account_opt: Option<H160> = FixedHashParser::<H160>::default()
-                    .from_matches_opt(m, "from-account", false)
-                    .or_else(|err| {
-                        let result: Result<Option<Address>, String> =
-                            AddressParser::new_sighash().from_matches_opt(m, "from-account", false);
-                        result
-                            .map(|address_opt| {
-                                address_opt.map(|address| {
-                                    H160::from_slice(&address.payload().args()).unwrap()
-                                })
-                            })
-                            .map_err(|_| err)
-                    })?;
-
-                let message = H256::from(blake2b_256(&binary));
-                let key_store_opt = from_account_opt
-                    .as_ref()
-                    .map(|account| (&*self.key_store, account));
-                let signature = sign_message(
-                    from_privkey_opt.as_ref(),
-                    key_store_opt,
-                    recoverable,
-                    &message,
-                )?;
-                let result = serde_json::json!({
-                    "message": format!("{:#x}", message),
-                    "signature": format!("0x{}", hex_string(&signature).unwrap()),
-                    "recoverable": recoverable,
-                });
-                Ok(result.render(format, color))
-            }
             ("sign-message", Some(m)) => {
-                let message: H256 =
-                    FixedHashParser::<H256>::default().from_matches(m, "message")?;
+                let magic_string = String::from("Nervos Message:");
+                let magic_bytes = magic_string.as_bytes();
+                let binary_opt = HexParser.from_matches_opt(m, "binary-hex", false)?;
+                let message_str_opt : Option<String> = NullParser.from_matches_opt(m, "message", false)?;
+
+                let to_sign : Vec<u8> = if let Some(binary) = binary_opt {
+                    binary
+                } else if let Some(message_str) = message_str_opt {
+                    message_str.as_bytes().to_vec()
+                } else {
+                    return Err(String::from("No value to sign"));
+                };
+                let msg_with_magic = [magic_bytes, &to_sign].concat();
```

Comment:
Have you considered the use case which don't want the magic bytes?
For example: a DApp want sign a message use `ckb-cli`.

Reply:
Are these two use cases different enough that they should be different subcommands? If there are messages with and without the magic byte it can potentially undermine the security benefits of this feature. Any information about this additional use case is helpful and appreciated!

Reply:
The point is the message need to be signed is from a CKB transaction generated by the DApp. Typically, the message is  from `blake2b(ckb_tx_hash + witnesses)`.

Reply:
Maybe better add a flag `--no-magic-bytes` for security consideration, then the default behave is changed, so it's a breaking change.

Reply:
For now I think it makes sense for us to merge this PR if we feel it adequately satisfies magic byte message signing. That way we can move forward with dev/QA of this feature on the Ledger device.

We should do another internal PR to account for this DApp use case and the `--no-magic-bytes` flag (after specification for it is finalized). Both of these features should be upstreamed to CKB-CLI in a single PR when finalized.

Reply:
I'm OK with the way move forward. Just one thing need to mention, I think the magic-bytes value deserve a `const` key word.

Reply:
We also consolidated sign-data with sign-message. Does that make sense to you?

Reply:
Yes, make sense to me.